core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -0,0 +1,199 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import './../../../common/rxjs-operators';
+import { AppSettings } from './../../../common/app-setting';
+
+import { WorkflowActionService } from './../workflow-graph/model/workflow-action.service';
+import { WorkflowGraph, WorkflowGraphReadonly } from './../workflow-graph/model/workflow-graph';
+import {
+  LogicalLink, LogicalPlan, LogicalOperator,
+  ExecutionResult, ErrorExecutionResult, SuccessExecutionResult
+} from './../../types/workflow-execute.interface';
+
+export const EXECUTE_WORKFLOW_ENDPOINT = 'queryplan/execute';
+
+
+/**
+ * ExecuteWorkflowService send the current workflow data to the backend
+ *  for execution, then receive backend's response to display on the
+ *  user interface. In this PR, the result will be printed onto console.
+ *  In the later PR, once the result panel is added, we will have a better
+ *  way to show the result of an execution.
+ *
+ * ExecuteWorkflowService will be responsible for transforming the frontend
+ *  formated graph into backend API formatted graph.
+ *
+ * Components should call executeWorkflow() function to fetch the current workflow
+ *  graph and execute the current workflow
+ *
+ * Components and Services should call getExecuteStartedStream() and subscribe
+ *  to the Observable in order to capture the time in which workflow graph
+ *  starts executing.
+ *
+ * Components and Services should call getExecuteEndedStream() and subscribe to
+ *  the Observable in order to capture the ending moment of execution and get
+ *  the final result of executing the workflow, either some results or errors
+ *
+ */
+@Injectable()
+export class ExecuteWorkflowService {
+
+  private executeStartedStream = new Subject<string>();
+  private executeEndedStream = new Subject<ExecutionResult>();
+
+  constructor(private workflowActionService: WorkflowActionService, private http: HttpClient) { }
+
+  /**
+   * Sends the current workflow data to the backend
+   *  to execute the workflow and get the results.
+   *
+   * @param workflowPlan
+   */
+  public executeWorkflow(): void {
+    // get the current workflow graph
+    const workflowPlan = this.workflowActionService.getTexeraGraph();
+
+    // create a Logical Plan based on the workflow graph
+    const body = ExecuteWorkflowService.getLogicalPlanRequest(workflowPlan);
```

Comment:
Why invoke class name and then call function? In rest of the code, local function is directly called.

Reply:
same as above